tests/integration/test_recrawl.py

```diff
@@ -0,0 +1,33 @@
+# vim:ts=4:sw=4:et:
+# Copyright 2016-present Facebook, Inc.
+# Licensed under the Apache License, Version 2.0
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+# no unicode literals
+
+import WatchmanTestCase
+import pywatchman
+
+
+@WatchmanTestCase.expand_matrix
+class TestRecrawl(WatchmanTestCase.WatchmanTestCase):
+
+    def test_failDontWaitForRecrawl(self):
+        root = self.mkdtemp()
+        self.touchRelative(root, '111')
+
+        self.watchmanCommand('watch', root)
+
+        # Force the next recrawl to have a delay
+        self.watchmanCommand('debug-delay-next-recrawl', root, 1000)
+
+        # Trigger the recrawl
+        self.watchmanCommand('debug-recrawl', root)
+
+        with self.assertRaises(pywatchman.WatchmanError):
```

Comment:
would be great to assert that the error message matches what you think it does; you can capture the context and then make an assertion on it, something along the lines of:

```
with self.assertRaises(pywatchman.WatchmanError) as ctx:
    ...
self.assertEquals("recrawl is active", str(ctx.exception))
```